components/screens/ledger-connect/LedgerConnectForSignFailed.tsx

```diff
@@ -0,0 +1,56 @@
+import { useNavigate, useSearchParams } from "react-router-dom";
+import Header from "@/components/GeneralHeader";
+import ledgerConnectFailedImage from "@/assets/images/ledger-connect-error.png";
+
+export default function LedgerConnectForSignFailed() {
+  const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
+  const redirect = searchParams.get("redirect");
+  const state = searchParams.get("state");
+
+  const retry = () => {
+    if (redirect) {
+      navigate(
+        {
+          pathname: redirect,
+        },
+        {
+          state: JSON.parse(decodeURIComponent(state ?? "") ?? "{}"),
```

Comment:
Do you have to pass data between full page screens and popup?

If not no need to serialize it. And you can pass the redirect inside the state.

Reply:
No, but it interacts across multiple pages: from the LedgerSend page to the LedgerSign page, then redirects to the Send page if successful or to the Failed page if unsuccessful. The state data in the Failed page comes from LedgerSign.

I decided to store state data in parameters because I plan to reuse this component for browser API signing.